### DISCUSSION TITLE
Improve search to make search filtering more exact.

### DISCUSSION THREAD

**Discussion by Pranav2612000** [outline/outline#6425] [Ideas]
Currently search allows for filtering of draft, active, or ALL documents(including archives). Users who only want to search a certain combination of documents are left without options. On large wiki installations this feature is needed to ensure hundreds of hits don't come up, only a smaller subset. Example: Trying to find a document that has a key phrase in the "archived" documents

**Implementation Ideas**
Currently the UI shows three options and allows you to choose one of the options. Perhaps a modal or some other UI element that allows a checkbox style selection between "Draft", "Active", "Archived" So you can choose any combination of the three! 

Please let me know what your thoughts are.
If you think this feature makes sense, I'll like to work on it ( In fact, I've already started 😅 )

**Comment by Pranav2612000** [comment]
@tommoor What are your opinions on this feature?

**Reply by tommoor** [reply]
Hi, open to it – but lets wait until https://github.com/outline/outline/pull/5814 merges (should be tomorrow) as it includes a large refactor of `SearchHelper` which will make this easier.

**Reply by Pranav2612000** [reply]
Hey @tommoor 
I can see that the PR has been merged. Can I start with the changes. The problem we are trying to solve is to allow the user to search document of a state that is not supported currently ( e.g just searching for archived or draft documents )
There are two approaches I could think of

Approach 1
- Update the dropdown options to
  - Published
  - Archived
  - Draft
- Change the radiobutton to a checkbox – i.e the user can select a multiple of these options.
This would allow the user to fine-tune and search for only the kind of documents he wants

Approach 2
- Keep the current options but add options for
  - Archived
  - Draft
These new options would allow the user to search in just archived and draft documents

Let me know what you think ( or if you have a different idea in mind ). I'll work on the backend changes depending on the approach we decide to take.

**Reply by tommoor** [reply]
Changing the existing UI to be multi-select makes the most sense to keep it flexible. You'll need to update `FilterOptions` to accept multiple selected values, and update the api endpoint. I think it's best to avoid the `includeX` style parameters that exist and add another filtering method – perhaps `statusFilter` to match the existing `dateFilter` param.

**Reply by Pranav2612000** [reply]
@tommoor I've created a WIP in progress [PR](https://github.com/outline/outline/pull/6537). Can you create a issue for this so that I can also link the PR to it.
Also feel free to review the PR and take a cursory look.
Here's how the API works -
It takes a `statusFilter` param which is an array of of `published`, `draft`, `archived`.
If status is `published` we show all documents with `publishedAt` timestamp ( and not having `archivedAt` timestamp )
If status is `draft` we show all documents with `publishedAt` timestamp as null ( and not having `archivedAt` timestamp )
If status is `archived` we show all documents having `archivedAt` at timestamp ( irrespective of `publishedAt` timestamp )

The query `or's` all the statuses so you can choose which statuses to show.
Let me know what you think.

**Reply by Pranav2612000** [reply]
Will also update the UI next to add the 3 options in a checkbox type dropdown.

**Comment by Pranav2612000** [comment]
@tommoor Bumping this up😅
